### PR TITLE
Add plugin to focus or unfocus selection

### PR DESCRIPTION
**Pull request by aerkiaga**
Signed-off-by: Aritz Erkiaga <aerkiaga3@gmail.com>

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

File: avogadro/qtplugins/focus/focus.cpp [hunk 1]
```diff
@@ -0,0 +1,111 @@
+/******************************************************************************
```
Comment (ghutchis) [inline avogadro/qtplugins/focus/focus.cpp:1]: Drop all this. As we edit files, we're going to a new top comment:
```
/******************************************************************************
  This source file is part of the Avogadro project.
  This source code is released under the 3-Clause BSD License, (see "LICENSE").
******************************************************************************/
```

(In particular, you're editing this in 2022, not 2012 .. and you're not working for Kitware.)

File: avogadro/qtplugins/focus/focus.h [hunk 2]
```diff
@@ -0,0 +1,71 @@
+/******************************************************************************
```
Comment (ghutchis) [inline avogadro/qtplugins/focus/focus.h:1]: Please use the new header:
```
/******************************************************************************
  This source file is part of the Avogadro project.
  This source code is released under the 3-Clause BSD License, (see "LICENSE").
******************************************************************************/
```

**Comment by ghutchis** [comment]
* I think it should be a bit more zoomed into the selection.

Load `1ayo` - select residue 10. Focus. Could probably be a bit more zoomed.

* It needs to call for a redraw after focus / unfocus:
```
    emit updateRequested();
```

* I also got it to totally make the molecule disappear. You have to be a bit careful with transforming the camera. Not sure if the camera matrix went sideways .. take a look at `resetview` for example.

File: avogadro/qtplugins/focus/focus.cpp (hunk 1 above)
Comment (aerkiaga) [reply avogadro/qtplugins/focus/focus.cpp:1]: Ooops... Yes, I'm sorry, I just copied the notice from a different file. I'm pretty stupid when it comes to legal code.

**Comment by aerkiaga** [comment]
Every thing done. The disappearing issue was due to a missed initialization, which should be fixed now. The current behavior is looking at the focused point from the current camera position, then scaling the view to put it at a fixed distance from the target.

**Comment by ghutchis** [comment]
Much better. One more minor thing. I think the zoom level should be estimated based on the radius / size of the selection.

Try this as an example. Download `1ayo` then select `ARG` (i.e., all the ARG residues). Some, but not a lot.

Focus. With this revision, you're zoomed in and can't even see the selection.

It's a minor thing, but I think if we get this right, it will be a very popular feature.

**Pull request merged**